### ISSUE TITLE
[$1000] Web - The invite message disappears if resized the window screen for copy pasted message

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#20929]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to staging dot on web chrome and go to any workspace
2. Go to invite members and on the invite message field, copy paste a long message multiple times
3. Now resize the screen to the small screen and notice that the message you copy pasted earlier disappears and the default message is shown
4. Now on the small screen only, copy paste a long message again and maximize the windows size to full screen
5. This time also, notice that the copy pasted texts disappeared when you switched to a larger screen

## Expected Result:
The invite message should not disappear if resized the window screen for copy pasted message

## Actual Result:
The invite message disappears if resized the window screen for copy pasted message
Platforms: web chrome

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.28.3

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/f03edc34-49cc-4609-a7f8-42e09829a8f4

https://github.com/Expensify/App/assets/93399543/7ec23bed-842c-4f15-b1ac-d79af1aa0b0f

**Expensify/Expensify Issue URL:**

**Issue reported by:** @priya-zha

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686374637223249

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0130acef183c3013c6</li>
        <li>Upwork Job ID: 1671582870354886656</li>
        <li>Last Price Increase: 2024-01-03</li>
    </ul>
</details>

**Comment by s-alves10** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When we send h_ello_, only _ello_ part is sent.

### What is the root cause of that problem?

The root cause is as in here. Remounting is the reason.
https://github.com/Expensify/App/issues/20522#issuecomment-1586360852

### What changes do you think we should make in order to solve the problem?

As in the above proposal, using onyx store to save temp data is a solution


### What alternative solutions did you explore? (Optional)

**Comment by adelekennedy** [comment]
reproduced on staging

**Comment by akinwale** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The welcome message on the invite page is reset when the window is resized.

### What is the root cause of that problem?
The default welcome note is being set in the componentDidUpdate method of the WorkspaceInviteMessagePage component.

### What changes do you think we should make in order to solve the problem?
Remove this line:
`this.setState({welcomeNote: this.getDefaultWelcomeNote()});`

### What alternative solutions did you explore? (Optional)
None.

**Comment by rain2o** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Any changes to the default invite message are lost when the screen size changes between mobile (small) and desktop (large) breakpoints.

### What is the root cause of that problem?

This is a result of how the app handles layouts. As described in [contributingGuides/NAVIGATION.md](https://github.com/Expensify/App/blob/main/contributingGuides/NAVIGATION.md#handling-wide-and-narrow-layouts):

> The wide and narrow layouts are conditionally rendered with different components in createResponsiveNavigator depending on screen size (isSmallScreen prop from the withWindowDimension.js).

All components within these layouts are unmounted/remounted when changing screen size between these specific breakpoints. The behavior described in the bug report is present in any input components that don't store the value in Onyx on change.

### What changes do you think we should make in order to solve the problem?


To be honest, I don't believe this would be considered a bug. Either that or it's an app-wide bug that impacts practically all components. I don't see this scenario being very common for typical users, so it seems like an edge case that will need to be handled as needed. 

### What alternative solutions did you explore? (Optional)

IF we decide this is a bug and we want to handle this per component, starting with the invite message component, then we would need to add `welcomeNote` to Onyx and update it `onChange` in the `TextInput` component.

**Comment by adelekennedy** [comment]
@abdulrahuman5196 any thoughts on the proposal above?

**Comment by abdulrahuman5196** [comment]
Thanks for the ping @adelekennedy. It seems to be an annoying bug.

@s-alves10 @rain2o 
Both rootcause is correct, the changes mentioned is almost same. I think saving in the onyx should be good.

But Form itself has inbuilt mechanism to store input values in onyx. Could you kindly check if its possible to use the same and update the proposals? 
Ref: https://github.com/Expensify/App/blob/main/contributingGuides/FORMS.md#form-drafts

**Comment by abdulrahuman5196** [comment]
Thank you for the updated proposal @s-alves10 , I will check if its working and update in a day.

**Comment by abdulrahuman5196** [comment]
This has become a generalized issue and seems to be there is some talks to solve it on app level - https://github.com/Expensify/App/issues/20404

**Comment by abdulrahuman5196** [comment]
Added a comment there to see how to proceed on this issue - https://github.com/Expensify/App/issues/20404#issuecomment-1613447879

**Comment by abdulrahuman5196** [comment]
Not overdue waiting on how the navigation refractor is proceeding

**Comment by mountiny** [comment]
Adding this on hold for #20404 where we will update how the navigation handles resize so no remount happens preventing bugs like this in its core

**Comment by adelekennedy** [comment]
still on hold

**Comment by adelekennedy** [comment]
Still holding here

**Comment by adelekennedy** [comment]
on hold

**Comment by adelekennedy** [comment]
still holding - moving to monthly

**Comment by adelekennedy** [comment]
still on hold

**Comment by abdulrahuman5196** [comment]
Seems long time since this issue was held on hold. Will check and update

**Comment by adelekennedy** [comment]
ah sorry @abdulrahuman5196 I never saw this comment! Updating

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)
